compilationunit.lua

```diff
@@ -226,6 +230,27 @@ function premake.extensions.compilationunit.getCompilationUnitDir(cfg)
 	return path.getabsolute(dir)
 end
 
+--
+-- Check whether the file should be excluded from the compilation unit
+--
+-- @param cfg
+--		The input configuration
+-- @param absfilename
+--		The filename of the file to check
+-- @return
+--		True if this should be excluded, False otherwise
+--
+function premake.extensions.compilationunit.isExcludedFromCU(cfg, fileName)
+	if cfg.compilationunitexclude then
+		for index, value in ipairs(cfg.compilationunitexclude) do
+			if path.getname(fileName) == value then
+				return true
```

Comment:
This could lead to problems when used in a project where you have a file with the same name in various subdirectories.